src/OpenTelemetry.Instrumentation.Runtime/RuntimeMetrics.cs

```diff
@@ -76,6 +76,9 @@ static RuntimeMetrics()
                 unit: "bytes",
                 description: "The amount of committed virtual memory for the managed GC heap, as observed during the latest garbage collection. Committed virtual memory may be larger than the heap size because it includes both memory for storing existing objects (the heap size) and some extra memory that is ready to handle newly allocated objects in the future. The value will be unavailable until garbage collection has occurred.");
 
+            MethodInfo mi = typeof(GC).GetMethod("GetGenerationSize", BindingFlags.NonPublic | BindingFlags.Static);
+            Func<int, ulong> getGenerationSize = mi.CreateDelegate<Func<int, ulong>>();
```

Comment:
as a defensive precaution it might be useful to avoid creating the gauge if the GetMethod call or the CreateDelegate call returns null. I don't know of any situation where that would happen but its good to be safe.